sdk/messaging/eventgrid/aznamespaces/shared_test.go

```diff
@@ -181,72 +179,75 @@ func newReceiverClient(t *testing.T, useSASKey bool) *aznamespaces.ReceiverClien
 }
 
 func newRecordingTransporter(t *testing.T, testVars testVars) policy.Transporter {
-	transport, err := recording.NewRecordingHTTPClient(t, nil)
-	require.NoError(t, err)
+	recordingOpts := &recording.RecordingOptions{
+		ProxyPort:    os.Getpid()%10000 + 20000,
+		UseHTTPS:     true,
+		TestInstance: t,
+	}
 
-	err = recording.Start(t, recordingDirectory, nil)
+	transport, err := recording.NewRecordingHTTPClient(t, recordingOpts)
 	require.NoError(t, err)
 
-	// err = recording.ResetProxy(nil)
-	// require.NoError(t, err)
+	err = recording.Start(t, recordingDirectory, recordingOpts)
+	require.NoError(t, err)
 
-	err = recording.AddURISanitizer(fakeTestVars.Endpoint, testVars.Endpoint, nil)
+	err = recording.AddURISanitizer(fakeTestVars.Endpoint, "https://[^/]+?/", recordingOpts)
```

Comment:
The proxy has a built-in URL sanitizer that replaces the first element of the host with `recording.SanitizedValue`. So if you make `fakeTestVars.Endpoint` something like `fmt.Sprintf("https://%s.eventgrid.azure.net", recording.SanitizedValue)` you wouldn't need this sanitizer.

Reply:
EventGrid has one extra bit with a region identifier. I could expand the default regex if that's something you think it should handle:

ex: <eventgrid-name>.eastus-1.eventgrid.azure.net

Reply:
(leaving this in for now. We can talk about a more generic version at another point)